docs/docs/dagster-basics-tutorial/6-custom-components.md

```diff
@@ -0,0 +1,109 @@
+
+## 2. Define the custom component
+
+When designing a component, keep its interface in mind. In this case, the assets that our component will create share the following attributes:
+
+- A DuckDB database shared across all assets.
+- A list of ETL assets, each with a URL path and a table name.
+
+The first step is to create a `dg.Model` for our ETL assets. These are similar to [Pydantic](https://docs.pydantic.dev/) models. This model will contain the two attributes that define an asset:
+
+<CodeExample
+  path="docs_snippets/docs_snippets/guides/tutorials/dagster-tutorial/src/dagster_tutorial/components/tutorial.py"
+  language="python"
+  startAfter="start_etl_model"
+  endBefore="end_etl_model"
+  title="src/etl_tutorial/components/tutorial.py"
+/>
+
+Next, add the interface to the `dg.Component` class. In this case, there will be a single attribute for the DuckDB database and a list of the `ETL` models we just defined:
+
+```python
+    duckdb_database: str
+    etl_steps: list[ETL]
+```
+
+The rest of the code will look very similar to the asset definitions we wrote earlier. The `build_defs` method constructs a `Definitions` object containing all the Dagster objects created by the component. Based on the interface defined at the class level, we will generate multiple ETL assets. The final Dagster object to include is the `resource` that the assets rely on, which can also be set via an attribute.
+
+<CodeExample
+  path="docs_snippets/docs_snippets/guides/tutorials/dagster-tutorial/src/dagster_tutorial/components/tutorial.py"
+  language="python"
+  startAfter="start_tutorial_component"
+  endBefore="end_tutorial_component"
+  title="src/etl_tutorial/components/tutorial.py"
+/>
+
+Run `dg check` to ensure that the component code is correct.
+
+## 3. Scaffold the component
+
+If we list our components again, we will see that the custom component is now registered:
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/tutorials/dagster-tutorial/commands/dg-list-components-custom.txt" />
+
+We can now scaffold it just like any other component:
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/tutorials/dagster-tutorial/commands/dg-scaffold-custom-component.txt" />
+
+This adds a new directory, `tutorials`, within `defs`:
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/tutorials/dagster-tutorial/tree/step-6b.txt" />
+
+## 4. Configure the component
+
+Set the configuration in the YAML file created when the component was scaffolded:
+
+<CodeExample
+  path="docs_snippets/docs_snippets/guides/tutorials/dagster-tutorial/src/dagster_tutorial/components/defs.yaml"
+  language="yaml"
+  title="src/dagster_tutorial/defs/tutorial/defs.yaml"
+/>
+
+:::info
+
+Before running `dg check` again, remove the `customers`, `orders`, and `payments` assets from `assets.py` and the `resource.py` file. The component is not responsible for generating these objects (otherwise there duplicate keys).
```

Comment:
I think this could use its own step, since not doing this will result in an error